xblock/fields.py

```diff
@@ -824,9 +833,23 @@ class String(JSONField):
     """
     MUTABLE = False
 
+    def _sanitize(self, value):
+        """
+        Remove the control characters that are not allowed in XML:
+        https://www.w3.org/TR/xml/#charsets
+        Leave all other characters.
+        """
+        if isinstance(value, unicode):
+            new_value = u''.join(ch for ch in value if unicodedata.category(ch)[0] != u'C' or ch in (u'\n', u'\r', u'\t'))
+        elif isinstance(value, str):
+            new_value = ''.join(ch for ch in value if ord(ch) >= 32 or ch in ('\n', '\r', '\t'))
+        else:
+            return value
+        return value if value == new_value else new_value
```

Comment:
But this PR changes that statement. If a string is stored in a String field and the string has control characters, it *is* mutated to remove the control characters. If it contains no control characters, then it remains unchanged.
I suppose the immutability of a stored string could be preserved if we `_santize`d the string on the way out to XML only. Would you prefer that to changing the immutability of String fields in this edge case?